### PR TITLE
[IoT Hub] Deprecated 'show-connection-string' command

### PR DESCRIPTION
**Pull request by anusapan**
**Description**  
Deprecating command **az iot hub show-connection-string**
image

Now, we have another command group for connection-string from IoT-Extension that can use instead
image

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from ._client_factory import iot_service_provisioning_factory
 from ._client_factory import iot_central_service_factory
 
-JOB_DEPRECATION_INFO = 'IoT Extension (azure-cli-iot-ext) Job commands'
+CS_DEPRECATION_INFO = 'IoT Extension (azure-cli-iot-ext) connection-string commands'
```
Comment (digimaun) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:11]: The IoT extension is now called "azure-iot". "azure-cli-iot-ext" is deprecated and should not be used anymore.

Comment (digimaun) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:11]: I think we should also show the equivalent command in the extension so the user doesn't have to look for it `az iot hub connection-string show`

Comment (anusapan) [reply src/azure-cli/azure/cli/command_modules/iot/commands.py]: Done

Comment (anusapan) [reply src/azure-cli/azure/cli/command_modules/iot/commands.py]: Done

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**